config/checkstyle-checks.xml

```diff
@@ -189,6 +189,13 @@
     <property name="message"
               value="Package comment marker should not be used if other visibility is defined"/>
   </module>
+  <module name="RegexpSingleline">
+    <property name="id" value="sinceVersionNumber"/>
+    <property name="format" value="\\* @since (?!8\.35$|8\.36$|8\.42$|9\.3$|3\.1$|5\.8$|3\.2$|5\.0$|3\.0$|6\.2$|8\.33$|8\.22$|5\.3$|5\.7$|3\.4$|4\.2$|6\.17$|6\.11$|6\.0$|6\.1$|6\.4$|8\.2$|6\.9$|6\.5$|8\.30$|8\.29$|5\.2$|7\.7$|5\.4$|7\.8$|4\.0$|6\.3$|6\.10$|6\.6$|8\.23$|8\.39$|3\.5$|6\.8$|4\.4$|8\.28$|6\.19$|8\.31$|8\.24$|7\.2$|8\.40$|6\.7$|7\.0$|7\.5$|8\.12$|3\.3$|7\.1$|5\.9$|8\.3$|8\.27$|8\.38$|8\.32$|5\.1$|8\.15$|8\.21$|8\.20$|8\.16$|6\.14$|8\.11$|8\.25$|6\.15$|8\.37$|6\.18$|8\.45$|5\.5$|7\.6$|8\.6$|8\.18$)([0-9]+)\.([0-9]+)$"/>
```

Comment:
Why we can not make regexp simple to read: 3 digits or list of known ?

Reply:
how to read 3 digits we want to violate 2 digits so any match of this `([0-9]+)\.([0-9]+)$` will be violation but we have to exculde list of know 2 digits

can you share example ?

Reply:
`/(([0-9])\.([0-9])\.([0-9]))|(10\.0)|(9\.12)$`

Please continue.
Play at https://regex101.com/

Reply:
Please see the new regex I like this more it is cleaner and no need for an ugly exclude list. It violates the version format X.Y when X >= 10 (10 is the current major version) 

```
@since 10.1   // violation x >= 10
@since 15.1    // violation X >= 10
@since 11  // violation X >= 10
@since 8.36   // ok X < 10
@since 15.1.0 // ok not X.Y
```

Reply:
> (([0-9])\.([0-9])\.([0-9]))|(10\.0)|(9\.12)$

according to our documentation. This will not violate when the file has at least one match so if there is one good format and one not good the file will pass

> Play at https://regex101.com/

Thank you for this it helps a lot I was looking for something similar